tony-core/src/main/java/com/linkedin/tony/ApplicationMaster.java

```diff
@@ -596,6 +599,11 @@ private boolean monitor() {
         break;
       }
 
+      if (this.untrackedTaskFailed) {
```

Comment:
I don't see you update final state based on this?

Reply:
That's being done in the [onTaskCompleted](https://github.com/linkedin/TonY/blob/master/tony-core/src/main/java/com/linkedin/tony/tensorflow/TonySession.java#L263) callback in the TonySession. 
In addition, it's also being updated again within the [monitor method](https://github.com/linkedin/TonY/blob/master/tony-core/src/main/java/com/linkedin/tony/ApplicationMaster.java#L626
), as soon as we exit the loop: